src/Microsoft.OData.Client/Serialization/HttpWebRequestMessage.cs

```diff
@@ -8,204 +8,187 @@ namespace Microsoft.OData.Client
+        /// </summary>
         public override string Method
         {
             get
             {
-                return this.effectiveHttpMethod;
+                return _effectiveHttpMethod;
             }
-
             set
             {
-                throw new NotSupportedException();
+                _requestMessage.Method = new HttpMethod(value);
             }
         }
 
-        /// <summary> Returns the collection of request headers. </summary>
-        public override IEnumerable<KeyValuePair<string, string>> Headers
+        /// <summary>
+        ///  Gets or set the credentials for this request.
+        /// </summary>
+        public override ICredentials Credentials
         {
             get
             {
-                List<KeyValuePair<string, string>> headerValues = new List<KeyValuePair<string, string>>(this.httpRequest.Headers.Count);
-                foreach (string headerName in this.httpRequest.Headers.AllKeys)
-                {
-                    string headerValue = this.httpRequest.Headers[headerName];
-                    headerValues.Add(new KeyValuePair<string, string>(headerName, headerValue));
-                }
-
-                return headerValues;
+                return _handler.Credentials;
             }
-        }
-
-        /// <summary> Returns the underlying HttpWebRequest </summary>
-        public System.Net.HttpWebRequest HttpWebRequest
-        {
-            get
+            set
             {
-                return this.httpRequest;
+                _handler.Credentials = value;
             }
         }
 
-        /// <summary>
-        /// Gets or set the credentials for this request.
-        /// </summary>
-        public override System.Net.ICredentials Credentials
-        {
-            get { return this.httpRequest.Credentials; }
-            set { this.httpRequest.Credentials = value; }
-        }
+
+#if !(NETCOREAPP1_0 || NETCOREAPP2_0)
 
         /// <summary>
         /// Gets or sets the timeout (in seconds) for this request.
         /// </summary>
         public override int Timeout
         {
-            get { return this.httpRequest.Timeout; }
-            set { this.httpRequest.Timeout = (int)Math.Min(Int32.MaxValue, new TimeSpan(0, 0, value).TotalMilliseconds); }
+            get
+            {
+                return (int)_client.Timeout.TotalSeconds;
+            }
+            set
+            {
+                _client.Timeout = new TimeSpan(0, 0, value);
+            }
         }
 
-        /// <summary>
-        /// Gets or sets the read and write timeout (in seconds) for this request.
-        /// </summary>
         public override int ReadWriteTimeout
         {
-            get { return this.httpRequest.ReadWriteTimeout; }
-            set { this.httpRequest.ReadWriteTimeout = (int)Math.Min(Int32.MaxValue, new TimeSpan(0, 0, value).TotalMilliseconds); }
+            get { throw new NotSupportedException(); }

```

Comment:
Maybe find something that would work for this as this may break existing clients.